e107_admin/plugin.php

```diff
@@ -665,51 +680,59 @@ private function pluginDownload()
 		$frm = e107::getForm();
 		$mes = e107::getMessage();
 		
+		e107::getDebug()->log("Plugin Download.");
+		$tmp = explode('&amp;', e_QUERY); // #1712 we need to process new ampersand separated query
+	  	$this->source = ($tmp[1]) ? $tmp[1] : "download";
+		parse_str($this->source); 
+		parse_str(base64_decode ($src)); 
+		$this->id         = !empty($tmp[1]) 	? intval($tmp[1]) 	: "";//mikeyfixed
+		$this->id         =  empty($this->id) 	? $plugin_id 		: $this -> id;//mikeyfixed
 	//	print_a($_GET); 	
-		
 		$string =  base64_decode($_GET['src']);	
+
 		parse_str($string, $data);
 
-		if(!empty($data['plugin_price']))
+		if(!empty($plugin_price))
```

Comment:
Check details here... 
 https://docs.google.com/document/d/1hbq645DZHW2Zy1oSos71yJMiqHbEk06BObM5HvzWNgQ

I'm back after the weekend :)